src/UI/EditorSurface.re

```diff
@@ -242,6 +250,9 @@ let%component make =
         )
       : 0.0;
 
+  let gutterWidth =
+    lineNumberWidth +. Constants.diffMarkerWidth +. Constants.gutterMargin;
```

Comment:
The lack of pretty much any kind of structure in this module makes it very difficult to know if every case has been covered. I'm hoping that just replacing the use of `lineNumberWidth` with this should do the trick, but I don't really think this is an improvement over what was either.

Ideally there would be separate "canvases" for separate parts, so that measures accumulate structurally rather than being pieced together from loose parts wherever needed. If there was a gutter component or function, for example, then the parts both available to it and exposed by it would be greatly reduced, and hence reduce the overall complexity. `lineNumberWidth` would most likely only be needed within the gutter, and the gutter width is probably the only measure needing to be exposed.

Reply:
Definitely - it'd be great to have the line numbers (and gutter) as a separate component. That would help significantly.